ddl/ddl_api.go

```diff
@@ -1505,7 +1505,14 @@ func buildTableInfoWithStmt(ctx sessionctx.Context, s *ast.CreateTableStmt, dbCh
 	if err = handleTableOptions(s.Options, tbInfo); err != nil {
 		return nil, errors.Trace(err)
 	}
-
+	if ctx.GetSessionVars().EnableClusteredIndex && !tbInfo.PKIsHandle && !config.GetGlobalConfig().AlterPrimaryKey {
```

Comment:
I think it's better to move to about L1267 `if constr.Tp == ast.ConstraintPrimaryKey ` and deal with the invisible index checking.